mesonbuild/minstall.py

```diff
@@ -417,7 +417,7 @@ def run_install_script(self, d):
             name = ' '.join(script + args)
             print('Running custom install script {!r}'.format(name))
             try:
-                rc = subprocess.call(script + args, env=child_env)
+                rc = subprocess.call([sys.executable] + script + args, env=child_env)
```

Comment:
This is actually a kind of workaround. With this to overcome the issue that some scripts need to have execute permissions to make the related tests run fine. See realted issue #4871

Reply:
That sounds like a bug elsewhere, we should be reading the chbang line and adding te necessary interpreter I think.

Reply:
This issue seems to be very deep inside the code somewhere. I read a lot about reading the shebang and adding the interpreter. I do not find the place in all the files with inheriting etc. It should be noted here that the above mentioned lines just solve this one specific permission denied issue. After checking out the git repo I get a whole lot of them until I chmod 755 the whole tree...

Reply:
The other problem is that if the script isn't python3 this is going to make it fall over. There's probably just something that needs to have it's conditions inverted. Let me look.

Reply:
Thanks a lot. Just let me sum up: The issue with the file permissions can just easily be shown on this example, it exists for really a lot of files in the tree.

Reply:
I'm reinstalling my openindiana VM (mine was i386 :)) And I'll have a look at a few of these.

Reply:
Cool. Please install also the gnome desktop which would need glib2.0. This will reveal a bug in the tests concerned with g-ir-scanner... I will need someones help there as I know how to do a workaround by setting an global env-var manually (->CC). I do not knpw how to do that with meson....

Reply:
I reverted this change and I don't see any other tests failing. There is some code that needs to be updated around install prefixes I think, I see the same failures on FreeBSD.

Reply:
So there is no issue with Solaris. Its a more general thing?